src/pretix/presale/views/organizer.py

```diff
@@ -719,6 +735,331 @@ def _events_by_day(self, before, after):
+            self.tz = pytz.timezone(next_ev.settings.timezone)
+            self.date = datetime_from.astimezone(self.tz).date()
+        else:
+            self.tz = self.request.organizer.timezone
+            self.date = now().astimezone(self.tz).date()
+
+    def _set_date(self):
+        if 'date' in self.request.GET:
+            self.tz = self.request.organizer.timezone
+            try:
+                self.date = dateutil.parser.parse(self.request.GET.get('date')).date()
+            except ValueError:
+                self.date = now().astimezone(self.tz).date()
+        else:
+            self._set_date_to_next_event()
+
+    def get(self, request, *args, **kwargs):
+        self._set_date()
+        return super().get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data()
+
+        before = datetime(
+            self.date.year, self.date.month, self.date.day, 0, 0, 0, tzinfo=UTC
+        ) - timedelta(days=1)
+        after = datetime(
+            self.date.year, self.date.month, self.date.day, 0, 0, 0, tzinfo=UTC
+        ) + timedelta(days=1)
+
+        ctx['date'] = self.date
+        ctx['cal_tz'] = self.tz
+        ctx['before'] = before
+        ctx['after'] = after
+
+        ctx['has_before'], ctx['has_after'] = has_before_after(
+            self.request.organizer.events.filter(
+                sales_channels__contains=self.request.sales_channel.identifier
+            ),
+            SubEvent.objects.filter(
+                event__organizer=self.request.organizer,
+                event__is_public=True,
+                event__live=True,
+                event__sales_channels__contains=self.request.sales_channel.identifier
+            ),
+            before,
+            after,
+        )
+
+        ebd = self._events_by_day(before, after)
+        if not ebd[self.date]:
+            return ctx
+
+        events = ebd[self.date]
+        shortest_duration = self._get_shortest_duration(events).total_seconds() // 60
+        # pick the next biggest tick_duration based on shortest_duration, max. 180 minutes
+        tick_duration = next((d for d in [5, 10, 15, 30, 60, 120, 180] if d >= shortest_duration), 180)
+
+        raster_size = min(self._get_raster_size(events), tick_duration)
+        events, start, end = self._rasterize_events(events, tick_duration=tick_duration, raster_size=raster_size)
+        calendar_duration = self._get_time_duration(start, end)
+        ctx["calendar_duration"] = self._format_duration(calendar_duration)
+        ctx['time_ticks'] = self._get_time_ticks(start, end, tick_duration)
+        ctx['start'] = datetime.combine(self.date, start)
+        ctx['raster_size'] = raster_size
+        # ctx['end'] = end
+        # size of each grid-column in calendar is based on shortest event duration
+        ctx['shortest_duration_rastered'] = next((d for d in [5, 10, 15, 30, 60, 90, 120, 180] if d >= shortest_duration), 180)
```

Comment:
Could we just re-use the tick_duration variable here or do I miss something?

Reply:
I used to have them different. tick_duration was just up to 60 minutes (just increased it recently for fullday-events to only include 3-hour steps) and shortest_duration_rastered basically just sets a class to size the grid-columns accordingly – which you just discovered a bug as I changed from fixed 5-minute raster to a flexible raster to fix the Firefox 1px overflow bug. So I will fix this.